### PR TITLE
Removes Apple mutation from Olives

### LINKED ISSUE #23665: Olive mutation
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

514.1589

### Issue Description

Mutated olives turning to ... an apple! Is it bug or feature

### What did you expect to happen?

I didnt expect THAT at all)

### What happened instead?

Wierd way to get apples

### Why is this bad/What are the consequences?

_No response_

### Steps to reproduce the issue.

mutating olives

### When did the problem start happening?

Not sure

### Extra information

_No response_

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by Mitchs98**
## What Does This PR Do
Makes Olives no longer mutate into anything, especially apples. My fault for copy pasting code and not noticing at all. Fixes #23665 

## Why It's Good For The Game
Olives shouldn't magically break biological and botanical barriers.

## Testing
1. Olive no mutate.

## Changelog
:cl:
fix: Olives no longer mutate into apples.
/:cl:

### CODE REVIEW

**Review by DGamerL — APPROVED** [review]
LGTM, could you check all other things that you added in case they also have stuff like this left?

File: code/modules/hydroponics/grown/olive.dm [hunk 1]
```diff
@@ -12,7 +12,7 @@
 	growthstages = 6
 	growing_icon = 'icons/obj/hydroponics/growing_fruits.dmi'
 	genes = list(/datum/plant_gene/trait/repeated_harvest)
-	mutatelist = list(/obj/item/seeds/apple)
+	mutatelist = list()
```
Comment (DGamerL) [inline code/modules/hydroponics/grown/olive.dm:15]:
```suggestion
```

**Review by DGamerL — CHANGES_REQUESTED** [review]

File: code/modules/hydroponics/grown/olive.dm (hunk 1 above)
Comment (Contrabang) [reply code/modules/hydroponics/grown/olive.dm:15]: IMO, I think its fine to leave it explicit in this case

Comment (DGamerL) [reply code/modules/hydroponics/grown/olive.dm:15]: Fine with me aswell

Comment (Mitchs98) [reply code/modules/hydroponics/grown/olive.dm:15]: I mirror'd other plants that don't have any other mutations when changing it like this.

**Comment by Mitchs98** [comment]
> LGTM, could you check all other things that you added in case they also have stuff like this left?

Yes I'll go through everything and check it as I have time and motivation to do so :D. I'll be sure to be more thorough with upcoming food port PRs too.

I'm very surprised no one reported olives magically turning into apples sooner.

**Review by Contrabang — APPROVED** [review]

**Review by Burzah — APPROVED** [review]

**Review by DGamerL — APPROVED** [review]

**Review by Sirryan2002 — APPROVED** [review]

**Comment by Sirryan2002** [comment]
Labeled with permission of headcoders

**Review by S34NW — APPROVED** [review]

**Pull request merged**